src/libmongoc/src/mongoc/mongoc-client.c

```diff
@@ -982,48 +982,35 @@ _mongoc_client_recv (mongoc_client_t *client,
 }
 
 
-/*
- *--------------------------------------------------------------------------
- *
- * mongoc_client_new --
- *
- *       Create a new mongoc_client_t using the URI provided.
- *
- *       @uri should be a MongoDB URI string such as "mongodb://localhost/"
- *       More information on the format can be found at
- *       http://docs.mongodb.org/manual/reference/connection-string/
- *
- * Returns:
- *       A newly allocated mongoc_client_t or NULL if @uri_string is
- *       invalid.
- *
- * Side effects:
- *       None.
- *
- *--------------------------------------------------------------------------
- */
 mongoc_client_t *
 mongoc_client_new (const char *uri_string)
 {
-   mongoc_topology_t *topology;
    mongoc_client_t *client;
-   mongoc_uri_t *uri;
+   bson_error_t error = {0};
 
+   if (!(client = mongoc_client_new_with_error (uri_string, &error))) {
```

Comment:
The [/Client/max_staleness](https://spruce.mongodb.com/task/mongo_c_driver_asan_ubuntu_debug_compile_asan_clang_openssl_patch_a203bafab7d412ed05f33b99229bacbc043c2268_61c6018d850e6143937a0bfb_21_12_24_17_21_19/logs?execution=0) test is failing due to this change. The test expects a log warning log on an invalid URI string with `mongoc_client_new`.

I do not think changing the log level is particularly harmful. I suggest updating the test to expect an error log level.

Reply:
`mongoc-uri.c` already logs warnings for unsupported values, so I think this may highlight a case where we're actually logging a redundant error message at a different log level. I'm not familiar with how `ASSERT_CAPTURED_LOG` works but I'll look into it and report back.